src/xrpld/app/tx/detail/InvariantCheck.cpp

```diff
@@ -1120,4 +1123,76 @@ ValidMPTIssuance::finalize(
         mptokensCreated_ == 0 && mptokensDeleted_ == 0;
 }
 
+//------------------------------------------------------------------------------
+
+void
+ValidPermissionedDomain::visitEntry(
+    bool,
+    std::shared_ptr<SLE const> const& before,
+    std::shared_ptr<SLE const> const& after)
+{
+    if (after->getType() != ltPERMISSIONED_DOMAIN)
+        return;
+    auto const& credentials = after->getFieldArray(sfAcceptedCredentials);
+    credentialsSize_ = credentials.size();
+    auto const sorted = credentials::makeSorted(credentials);
+    isUnique_ = !sorted.empty();
+
```

Comment:
`isSorted_` is initialized to `false` (a few lines below). As long as the the input `credentials` has at least one element, `isSorted_` must be updated with the correct value.

Reply:
And credentials have no elements, I would expect this to be logged
```
    if (!isUnique_)
    {
        JLOG(j.fatal()) << "Invariant failed: permissioned domain credentials "
                           "aren't unique";
```
... but in fact, because of how `isSorted_` is initialised, this will be logged instead:
```
    if (!isSorted_)
    {
        JLOG(j.fatal()) << "Invariant failed: permissioned domain credentials "
                           "aren't sorted";
```

That's what I meant above by "misleading logging"

Reply:
NVM, I was wrong, if `credentials` is empty, this will be logged
```
    if (!credentialsSize_)
    {
        JLOG(j.fatal()) << "Invariant failed: permissioned domain with "
                           "no rules.";
```

Reply:
Ah I see you swapped the logging. That makes it easier to understand - all good.